machine_data_model/nodes/connectors/opcua_connector.py

```
@@ -0,0 +1,440 @@
+        :param kwargs: method arguments expressed as key/name - value pairs
+        :return: dict of results in the form of name - value pairs
+        """
+        path_parts = path.lstrip('/').split("/")
+        input_arg_path = path + '/InputArguments'
+        # TODO: this might fail if the method as no input parameters
+        method_inputs = await self._async_get_remote_node(input_arg_path)
+        if method_inputs is None:
+            return None
+
+        inputs = await method_inputs.read_value() # returns a list of Argument-Class
+        assert isinstance(inputs, list), "inputs must be a list"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)